### DISCUSSION TITLE
Return Periods Computation for River Flood

### DISCUSSION THREAD

**Discussion by Mdf196** [CLIMADA-project/climada_python#906] [General]
Hello everyone, I have a question regarding the issue described in the following topic: https://github.com/CLIMADA-project/climada_python/issues/643. 

The question is: 
While using ISIMIP 2b study data (approximately 95 projected years), is it possible to calculate the flood impact with a 200-year return period by modifying the annual event frequency from 1/95 to 1/200 ? Is this conceptually correct? The mentioned topic says that I can compute up to a maximum of 95 years impact return period.

For example:

`rf = RiverFlood.from_nc(countries = ['Select_your_country'],
                                        origin = False,
                                        years = [2006,2007,.......,2100], 
                                        dph_path = "path_to_your_file", 
                                        frc_path = "path_to_your_file",
                                        ISINatIDGrid = True,)

for i in range(len(rf.frequency)):
    rf.frequency[i] = 0.005
rf.frequency`

In this way, using the calc_freq_curve() and local_exceedance_imp command, it's possible to plot and computer either the EFC and the impact with a 200-year return period, without having always the same exceedance impact values for return periods > 95 years, as the mentioned topic's issue. 
But, is it formally correct?

Thank you in advance for the clarifications.

**Comment by bguillod** [comment]
Hi @Mdf196 ,

What you are doing is conceptually wrong, as you are altering the underlying data that way.

However, isimip2b flood data contains simulations from multiple models and thereby significantly more than 95 years of data. I would recommend you to consider the following points:
- Rather than using all isimip2b data, use a subset of years, e.g. 2040-2060 to target 2050 or 2080-2099 to target end of the century. Or at least it is not clear to me what you are trying to achieve using the full 2005-2099 data.
- Read data from all model combinations, which provides you a much stronger data basis and will enable you to estimate higher return periods.
- be aware of recent discussions on return period estimation, such as #904.

I hope this helps

**Reply by Mdf196** [reply]
Thank you so much for the explanation, it was very helpful.

Regarding point 1, at the moment I'm just playing with the data and the code itself, but I did exactly what you mentioned (subsetting and computing impact). By the way I was interested to see a complete (and correct ) exceedance frequency curve (and exceedance impacts) with longer return periods, just to understand the behavior of these elements.

Regarding point 2-3, thank you for the advices. I encountered the same issue while computing exceedance impact with both commands (localc_exceedance_imp and calc_frec_curve.impact), and I was trying to figure it out why the exceedance impacts were different.

Edit:
@bguillod I take this opportunity to ask you another question. 
Regarding the frequency assigned to the events present in the .hdf5 files available in the Desktop version of Climada ("river_flood_150arcsec_rcp26_ITA_2010/30/50_2030/50/70"), how is it possible that these files can be used to estimate the impact for return periods of up to 500 years, if they cover a shorter time span? 
This topic https://github.com/CLIMADA-project/climada_python/issues/526 explains how the frequency is calculated, which is equal to 1/480, but how is it possible to obtain such a value if the maximum time span is 2070?
Thank you again for the answer.

**Reply by bguillod** [reply]
> Regarding the frequency assigned to the events present in the .hdf5 files available in the Desktop version of Climada ("river_flood_150arcsec_rcp26_ITA_2010/30/50_2030/50/70"), how is it possible that these files can be used to estimate the impact for return periods of up to 500 years, if they cover a shorter time span?
This topic https://github.com/CLIMADA-project/climada_python/issues/526 explains how the frequency is calculated, which is equal to 1/480, but how is it possible to obtain such a value if the maximum time span is 2070?

Where does the assumption come from that you can use these data to estimate return periods up to 500 years?

I am not sure I fully understand your question, but my feeling is that it is about theoretical background more than about climada. The maximum time span has no relation to the number of simulated years. Any calendar year can be simulated multiple times to sample possible weather events. In fact each year is simulated 24 times (4 times 6 according to the post in that topic), so with e.g. 20 years you actually get 480 simulated years. This is not 480 years in the future but 480 possible realizations of yearly flood events to try to sample the distribution and hence extreme events. It's an attempt to sample random weather that could occur within a certain climate state. This enables you to estimate rare events, although values for high return periods (e.g., 100 years) might not be robustly estimated (in other words, the exact value is very inaccurately estimated due to the small sample size). The return period estimation in climada silently returns a value, but the user needs expertise in order to assess the robustness of the value.

**Reply by Mdf196** [reply]
Ok, thank you for the explanation. This is what I was looking for to solve my doubts.
